### ISSUE TITLE
Fix error in the log when go to /admin

### ISSUE THREAD

**Issue by fmontes** [dotCMS/core#16543]
### Steps to reproduce
1. Logout
2. Remove all your cookies
3. Go to http://localhost:8080/admin
4. See in catalina out the error:

### Error:

```
14:31:41.006  ERROR listeners.SwitchSiteListener - User null does not have permission to host:demo.dotcms.com
com.dotmarketing.exception.DotSecurityException: User null does not have permission to host:demo.dotcms.com
    at com.dotmarketing.portlets.contentlet.business.HostAPIImpl.find_aroundBody6(HostAPIImpl.java:289) ~[classes/:?]
    at com.dotmarketing.portlets.contentlet.business.HostAPIImpl$AjcClosure7.run(HostAPIImpl.java:1) ~[classes/:?]
    at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
    at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[classes/:?]
    at com.dotcms.aspects.interceptors.CloseDBIfOpenedMethodInterceptor.invoke(CloseDBIfOpenedMethodInterceptor.java:29) ~[classes/:?]
    at com.dotcms.aspects.aspectj.CloseDBIfOpenedAspect.invoke(CloseDBIfOpenedAspect.java:41) ~[classes/:?]
    at com.dotmarketing.portlets.contentlet.business.HostAPIImpl.find(HostAPIImpl.java:266) ~[classes/:?]
    at com.dotcms.listeners.SwitchSiteListener.triggerSwitchSiteEvent(SwitchSiteListener.java:57) ~[classes/:?]
    at com.dotcms.listeners.SwitchSiteListener.attributeAdded(SwitchSiteListener.java:31) ~[classes/:?]
    at org.apache.catalina.session.StandardSession.setAttribute(StandardSession.java:1522) ~[catalina.jar:8.5.32]
    at org.apache.catalina.session.StandardSession.setAttribute(StandardSession.java:1415) ~[catalina.jar:8.5.32]
    at org.apache.catalina.session.StandardSessionFacade.setAttribute(StandardSessionFacade.java:137) ~[catalina.jar:8.5.32]
    at com.dotmarketing.servlets.LoginEditModeServlet.service(LoginEditModeServlet.java:89) ~[classes/:?]
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:742) ~[servlet-api.jar:?]
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) ~[catalina.jar:8.5.32]
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
    at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52) ~[tomcat-websocket.jar:8.5.32]
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
    at com.dotmarketing.filters.CMSFilter.doFilterInternal(CMSFilter.java:199) ~[classes/:?]
    at com.dotmarketing.filters.CMSFilter.doFilter(CMSFilter.java:56) ~[classes/:?]
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
```

### Possible Solution
Per @wezell the problem is in this file: https://github.com/dotCMS/core/blob/master/dotCMS/src/main/java/com/dotmarketing/servlets/LoginEditModeServlet.java

## Acceptance Criteria
1. Not show error in the log under this conditions
2. Work in all the supported browsers (don't forget IE11 and iPad)
3. Multilanguage
4. Unit test

**Comment by bryanboza** [comment]
Fixed, tested on master // Postgres // FF

**Comment by wezell** [comment]
Fixed here: #16605
